sys/base64/base64.c

```diff
@@ -232,34 +229,44 @@ int base64_decode(const void *base64_in, size_t base64_in_size,
         return BASE64_ERROR_BUFFER_OUT;
     }
 
-    int iterate_data_buffer = 0;
-    uint8_t n_num = 0;
-    int nLst = getcode(in[0]) << 2;
-    int code = 0;
-
-    int mask = 2;
-
-    for (int i = 1; i < (int)(base64_in_size); i++) {
-        code = getcode(in[i]);
-
-        if (code == BASE64_NOT_DEFINED || code == BASE64_EQUALS) {
-            continue;
+    const uint8_t *end = base64_in + base64_in_size;
+    uint8_t codes[4];
+    size_t n_codes;
+
+    while (1) {
+        n_codes = 0;
+        do {
+            if (in == end) {
+                switch (n_codes) {
+                    case 0:
+                        /* nothing to do */
+                        break;
+                    case 1:
+                        return BASE64_ERROR_DATA_IN_SIZE;
+                    case 2:
+                        codes[2] = codes[3] = 0;
+                        decode_four_codes(out, codes);
+                        out += 1;
+                        break;
+                    case 3:
+                        codes[3] = 0;
+                        decode_four_codes(out, codes);
+                        out += 2;
+                        break;
+                }
+                *data_out_size = (size_t)out - (size_t)data_out;
```

Comment:
```suggestion
                *data_out_size = (uintptr_t)out - (uintptr_t)data_out;
```

makes no difference, but easier to understand.